src/app/components/coupon-profile/coupon-profile.component.ts

```diff
@@ -15,6 +15,9 @@ import { Component, OnInit } from '@angular/core';
 })
 export class CouponProfileComponent implements OnInit {
   coupon: Coupon;
+  editable = false;
+  purchasable = false;
+  cancelable = false;
```

Comment:
LOVE that idea, I think its exactly how it should be done, but I think we should show all of them together, and just make the button disabled if the flag is false.